subprojects/core/src/main/java/org/gradle/internal/buildtree/DefaultBuildTreeLifecycleController.java

```diff
@@ -82,9 +82,16 @@ public <T> T fromBuildModel(boolean runTasks, BuildTreeModelAction<? extends T>
         return runBuild(() -> {
             modelCreator.beforeTasks(action);
             if (runTasks && isEligibleToRunTasks()) {
-                ExecutionResult<Void> result = workController.scheduleAndRunRequestedTasks(null);
```

Comment:
❓ I was looking in to this, how to make this nicer, and came with this proposal:
https://github.com/gradle/gradle/pull/35836

WDYT?